integration-tests/chopsticks/src/tests/index.ts

```diff
@@ -0,0 +1,59 @@
+import { beforeAll, afterAll } from 'vitest'
+import { connectParachains, connectVertical } from '@acala-network/chopsticks'
+import { setTimeout } from 'timers/promises'
+
+import * as SpiritnetConfig from '../network/spiritnet.js'
+import * as PolkadotConfig from '../network/polkadot.js'
+import * as HydraDxConfig from '../network/hydraDx.js'
+import type { Config } from '../network/types.js'
+import { setStorage } from './utils.js'
+
+export let spiritnetContext: Config
+export let hydradxContext: Config
+export let polkadotContext: Config
+
+beforeAll(async () => {
+	spiritnetContext = await SpiritnetConfig.getContext()
+	hydradxContext = await HydraDxConfig.getContext()
+	polkadotContext = await PolkadotConfig.getContext()
+
+	// Setup network
+	await connectVertical(polkadotContext.chain, spiritnetContext.chain)
+	await connectVertical(polkadotContext.chain, hydradxContext.chain)
+	await connectParachains([spiritnetContext.chain, hydradxContext.chain])
+
+	const newBlockConfig = { count: 2 }
+	// fixes api runtime disconnect warning
+	await setTimeout(50)
+	// Perform runtime upgrade and establish xcm connections.
```

Comment:
This will need to go after we go live, right? Same for some of the HydraDX storage overrides. Do you think we should point the tests to a specific block on all the three chains, to reduce even more externalities?

Reply:
That's a good point. I am not sure. I think having the latest chain state has more value than having consistent tests. So I would propose to update these tests, once the referendum from HydraDx is through. We would also notice here, if they have changed something on their side. What do you think?

Reply:
Yeah but I feel this approach is not maintainable. What if we have tests for 10 different chains, 6 of which must be updated after our runtime upgrade, and 2 after they have upgraded their runtime? I am not sure there is an easy solution here. At the very least, I would probably separate e2e tests by runtime couples, something like (spiritnet-1.12.1 <-> hydra-1) another for (spiritnet-1.13.0 <-> hydra-1), and another one for (spiritnet-1.13.0 <-> hydra-2). Then we know exactly when each test becomes deprecated and can be removed. Let's make a ticket for investigating how to best solve this issue, and maybe paste the suggestion I just gave. We can have a proper discussion once we will tackle the issue properly.

Reply:
> Currently, I am afraid that we might spend time solving a problem that may not arise in the next 1, 2, or even 4 years.
I don't actually think that's true. If anything, now that we have a "more open" XCM configuration, we might get more inbound requests to support other use cases, and we definitely need tests for that. Anyway, as long as we have a ticket, this conversation can be resolved. But please, make another ticket to remove any stuff that is not relevant after our first runtime upgrade goes through, after HDX first proposal goes through, our second proposal goes through, and HDX second proposal goes through 😁

Reply:
I hope I am wrong. This would be great for KILT :100:. And I totally agree, we should test each interaction with another chain. I created a ticket to update the storage mocks, once the HydraDx referendum is through: [#3344](https://github.com/KILTprotocol/ticket/issues/3344)